src/DataAccess/Hydrators/CategoriesHydrator.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Fsbe\DataAccess\Hydrators;
+
+use Fsbe\DataAccess\Database;
+use Fsbe\Entities\Category;
+use Fsbe\Entities\Categories;
+
+class CategoriesHydrator
+{
+    public static function hydrateFromDb(Database $db): Categories
+    {
+        $sql = 'SELECT `products`.`category_id` As `id`, `categories`.`name`, COUNT(`category_id`) AS `products` '
+               . 'FROM `products` '
+               . 'JOIN `categories` '
+               . 'WHERE `products`.`category_id` = `categories`.`id` '
+               . 'GROUP BY 1;';
+
+        $statement = $db->getConnection()->prepare($sql);
+
+        $statement->execute();
+
+        $statement->setFetchMode(\PDO::FETCH_CLASS, Category::class);
+
+        $result = $statement->fetchAll();
+
+        $categories = new Categories();
```

Comment:
more a comment than anything - this could be improved by sticking to Dependancy injection and passing this in as a param rather than instansiating it inside the method (tightly coupled).